geode-core/src/main/java/org/apache/geode/internal/metrics/GeodeCompositeMeterRegistry.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.internal.metrics;
+
+import java.util.Set;
+
+import io.micrometer.core.instrument.binder.MeterBinder;
+import io.micrometer.core.instrument.composite.CompositeMeterRegistry;
+
+public class GeodeCompositeMeterRegistry extends CompositeMeterRegistry {
+
+  private final Set<MeterBinder> binders;
+
+  public GeodeCompositeMeterRegistry(Set<MeterBinder> binders) {
+    super();
+    this.binders = binders;
+  }
+
+  void registerBinders() {
```

Comment:
I'm not thrilled that the configuration is split between this method (to register the binders) and the factory (to configure everything else). Splitting the configuration seems troublesome. I have an idea that I'm also not sure I like: Let the factory do _all_ of the configuration (including binding the binders) and pass `Collection<Closeable>`  to this class's constructor. Then let this class simply close the `Closeable`s.